packages/effect/src/Stream.ts

```diff
@@ -4329,6 +4329,44 @@ export const zipLatest: {
   <A, E, R, A2, E2, R2>(self: Stream<A, E, R>, that: Stream<A2, E2, R2>): Stream<[A, A2], E | E2, R | R2>
 } = internal.zipLatest
 
+/**
+ * Zips multiple streams so that when a value is emitted by any of the streams,
+ * it is combined with the latest values from the other streams to produce a result.
+ *
+ * Note: tracking the latest value is done on a per-chunk basis. That means
+ * that emitted elements that are not the last value in chunks will never be
+ * used for zipping.
+ *
+ * @example
+ * import { Stream } from "effect"
+ *
+ * const stream = Stream.zipLatestAll(
+ *     Stream.fromSchedule(Schedule.spaced('500 millis')),
+ *     Stream.fromSchedule(Schedule.spaced('1000 millis')),
+ *     Stream.fromSchedule(Schedule.spaced('2000 millis')),
+ * ).pipe(Stream.tap(Console.log))
+ *
+ * Effect.runPromise(Stream.runDrain(stream))
+ * // Output:
+ * [ 0, 0, 0 ]
+ * [ 1, 0, 0 ]
+ * [ 1, 1, 0 ]
+ * [ 2, 1, 0 ]
+ * [ 3, 1, 0 ]
+ * [ 3, 1, 1 ]
+ * // .....
+ *
+ * @since 2.0.0
+ * @category zipping
+ */
+export const zipLatestAll: <T extends Array<Stream<unknown>>>(
+  ...streams: T
+) => Stream<
+  internal.ZipLatestAllValues<T>,
```

Comment:
Type helpers should not be internal, as they are stripped from the final build.

Also, I think these types should just be inlined :)

```suggestion
  T extends ReadonlyArray<Stream<infer _R, infer _E, infer _A>> ? _R : never,
```

Will need to test if this works with empty arguments too.

Reply:
This is the suggestion that i don't understand, because AFAIU
`T extends ReadonlyArray<Stream<infer _R, infer _E, infer _A>> ? _A : never,` is not the same as `ZipLatestAllValues`.

I am also not sure what to do with the type helpers, because if we choose to inline it, we would need to do it in 2 both -`Stream.ts` and `internal/stream.ts`. These types will look pretty heavy inlined. Is it ok? What do you think about it?